src/Application/Migrations/24_function_ticket_reset.sql

```diff
@@ -0,0 +1,76 @@
+BEGIN;
+
+--param_handle_id is used for the log entry
+CREATE OR REPLACE FUNCTION ticket_reset(param_ticket_id bigint)
+  RETURNS TABLE (ticket_id bigint,  from_state enum_ticket_state, to_state enum_ticket_state, follow_meta_ticket boolean, follow_encoding_ticket boolean) AS
+$$
+DECLARE
+  ticket tbl_ticket%rowtype;
+  inital_state enum_ticket_state;
+BEGIN
+  --first get the full given ticket
+  SELECT * INTO ticket FROM tbl_ticket where id = param_ticket_id;
+
+  IF NOT FOUND THEN
+    RAISE WARNING 'ticket with id % does not exist', param_ticket_id;
+    RETURN;
+  END IF;
+
+  --reset the given ticket to inital state
+  inital_state := ticket_state_initial(ticket.project_id, ticket.ticket_type);
+  IF inital_state <> ticket.ticket_state THEN
+    ticket_id := ticket.id;
+    from_state := ticket.ticket_state;
+    to_state := inital_state;
+    follow_meta_ticket := FALSE;
+    follow_encoding_ticket := FALSE;
+    RETURN NEXT;
```

Comment:
I guess the UPDATE query should happen before the next row is returned, otherwise log entries might be created noting state changes which might not have happened due to errors?!
(same 2 times below)